### PR TITLE
Updated the Contract WebaverseERC721

### PR DESCRIPTION
**Pull request by mohsinanees**
I have updated the smart contract WebaverseERC721 with the following changes:

1- Added `mintBatch` function that could allow us to mint tokens in bulk.
2- Resolved the token URI issues in the previous one by adding the check for the provided URI in the mint function.
3- Redeployed the contract on rinkeby and updated the ABI and config on redis.
4- Added events URI, Claim, ExternalClaim to enhance the speed and efficiency of fetching the data from the smart contract on the redis-server and api-backend.

### CODE REVIEW

**Review by avaer — COMMENTED** [review]
Why is fs in package.json? That is a native module.

**Review by avaer — COMMENTED** [review]
Should not put your own dev machine directory names into the configs…

**Review by avaer — CHANGES_REQUESTED** [review]
Very messy code needs cleanup, but largely ok.

**Comment by avaer** [comment]
Why is hardhat labeled a type of network in the configs?

**Comment by avaer** [comment]
Why is the etherscan api key leaked?

File: tsconfig.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     // "sourceMap": true,                           /* Generates corresponding '.map' file. */
     // "outFile": "./",                             /* Concatenate and emit output to single file. */
     "outDir": "lib",                                /* Redirect output structure to the directory. */
-    "rootDir": "src",                               /* Specify the root directory of input files. Use to control the output directory structure with --outDir. */
+    "rootDir": "/home/mohsin/Desktop/Webaverse-repos/drop-contracts",                               /* Specify the root directory of input files. Use to control the output directory structure with --outDir. */
```
Comment (abeersaqib) [inline tsconfig.json:18]: @mohsinanees Please resolve this

**Comment by mohsinanees** [comment]
> Why is hardhat labeled a type of network in the configs?

It is labeled as a network because hardhat runs its own virtual VM to run the tests just like the ganache-cli.

**Review by avaer — APPROVED** [review]

**Pull request merged**